effekt/shared/src/main/scala/effekt/generator/chez/ChezSchemeLift.scala

```diff
@@ -99,17 +99,17 @@ object ChezSchemeLift extends Backend {
 
     case Hole => chez.Builtin("hole")
 
-    case State(id, init, region, body) if region == symbols.builtins.globalRegion =>
+    case State(id, init, stateTpe, region, body) if region == symbols.builtins.globalRegion =>
       chez.Let(List(Binding(nameDef(id), chez.Builtin("box", toChez(init)))), toChez(body))
 
-    case State(id, init, region, body) =>
+    case State(id, init, stateTpe, region, body) =>
       chez.Let(List(Binding(nameDef(id), chez.Builtin("fresh", Variable(nameRef(region)), toChez(init)))), toChez(body))
 
     case Try(body, tpe, handler) =>
       val handlers: List[chez.Handler] = handler.map { h =>
-        val names = RecordNames(h.id)
+        val names = RecordNames(h.id.symbol)
```

Comment:
I previously tried to get rid of the dependency on symbols in all phases after core. Why is this necessary here?